backend/src/apiserver/main.go

```diff
@@ -195,13 +234,35 @@ func startHttpProxy(resourceManager *resource.ResourceManager) {
 	// Register a handler for Prometheus to poll.
 	topMux.Handle("/metrics", promhttp.Handler())
 
-	http.ListenAndServe(*httpPortFlag, topMux)
+	if tlsConfig != nil {
+		glog.Info("Starting Https Proxy")
+		https := http.Server{
+			TLSConfig: tlsConfig,
+			Addr:      *httpPortFlag,
+			Handler:   topMux,
+		}
+		https.ListenAndServeTLS("", "")
+	} else {
+		glog.Info("Starting Http Proxy")
+		http.ListenAndServe(*httpPortFlag, topMux)
+	}
+
 	glog.Info("Http Proxy started")
 }
 
-func registerHttpHandlerFromEndpoint(handler RegisterHttpHandlerFromEndpoint, serviceName string, ctx context.Context, mux *runtime.ServeMux) {
+func registerHttpHandlerFromEndpoint(handler RegisterHttpHandlerFromEndpoint, serviceName string, ctx context.Context, mux *runtime.ServeMux, tlsConfig *tls.Config) {
 	endpoint := "localhost" + *rpcPortFlag
-	opts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(math.MaxInt32))}
+	var opts []grpc.DialOption
+	if tlsConfig != nil {
+		// local client connections to grpc should not require tls
```

Comment:
by local, do you mean from the http proxy?

Reply:
there instances where the grpc server within api server communicates with the local http server that it also creates as a separate routine, so local as in these are part of the same container,